src/SFA.DAS.Vacancies/Application/Vacancies/Queries/GetTraineeshipVacanciesQueryHandler.cs

```diff
@@ -0,0 +1,90 @@
+        private readonly IAccountLegalEntityPermissionService _accountLegalEntityPermissionService;
+        private readonly ICourseService _courseService;
+        private readonly VacanciesConfiguration _vacanciesConfiguration;
+
+        public GetTraineeshipVacanciesQueryHandler(IFindTraineeshipApiClient<FindTraineeshipApiConfiguration> findTraineeshipApiClient, 
+            IAccountLegalEntityPermissionService accountLegalEntityPermissionService, 
+            ICourseService courseService,
+            IOptions<VacanciesConfiguration> vacanciesConfiguration)
+        {
+            _findTraineeshipApiClient = findTraineeshipApiClient;
+            _accountLegalEntityPermissionService = accountLegalEntityPermissionService;
+            _courseService = courseService;
+            _vacanciesConfiguration = vacanciesConfiguration.Value;
+        }
+
+        public async Task<GetTraineeshipVacanciesQueryResult> Handle(GetTraineeshipVacanciesQuery request, CancellationToken cancellationToken)
+        {
+            if (!string.IsNullOrEmpty(request.AccountLegalEntityPublicHashedId))
+            {
+                switch (request.AccountIdentifier.AccountType)
+                {
+                    case AccountType.Unknown:
+                        throw new SecurityException();
+                    case AccountType.External:
+                        request.AccountLegalEntityPublicHashedId = string.Empty;
+                        request.AccountPublicHashedId = string.Empty;
+                        break;
+                    default:
+                    {
+                        var accountLegalEntity = await _accountLegalEntityPermissionService.GetAccountLegalEntity(request.AccountIdentifier,
+                                request.AccountLegalEntityPublicHashedId);
+                        
+                        if (accountLegalEntity == null)
+                        {
+                            throw new SecurityException();
+                        }
+                        break;
+                    }
+                }
+            }
+
+            var categories = _courseService.MapRoutesToCategories(request.Routes);
+
+            var vacanciesTask = _findTraineeshipApiClient.Get<GetTraineeshipVacanciesResponse>(new GetTraineeshipVacanciesRequest(
+                request.PageNumber, request.PageSize, request.AccountLegalEntityPublicHashedId, 
+                request.Ukprn, request.AccountPublicHashedId, request.RouteId, request.NationWideOnly, 
+                request.Lat, request.Lon, request.DistanceInMiles, categories, request.PostedInLastNumberOfDays, request.Sort));
+
+            await Task.WhenAll(vacanciesTask);
+
+            foreach (var vacanciesItem in vacanciesTask.Result.TraineeshipVacancies)
+            {
+                if (vacanciesItem.RouteId == null)
+                {
+                    continue;
+                }
```

Comment:
This isn't required. Everything should have a routeId that is going to be indexed. If it doesnt then we should still add the VacancyUrl for it